data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/CsaProfileCrac.java

```diff
@@ -28,26 +26,50 @@ public class CsaProfileCrac implements NativeCrac {
 
     private final QueryCatalog queryCatalogCsaProfileCrac;
 
+    private Map<String, Set<String>> keywordMap;
+
     public CsaProfileCrac(TripleStore tripleStoreCsaProfileCrac) {
         this.tripleStoreCsaProfileCrac = tripleStoreCsaProfileCrac;
         this.queryCatalogCsaProfileCrac = new QueryCatalog(CsaProfileConstants.SPARQL_FILE_CSA_PROFILE);
+        this.keywordMap = new HashMap<>();
     }
 
     @Override
     public String getFormat() {
         return "CsaProfileCrac";
     }
 
+    public void clearContext(String context) {
+        tripleStoreCsaProfileCrac.clear(context);
+    }
+
+    public void fillKeywordMap(Map<String, Set<String>> keywordMap) {
+        this.keywordMap = keywordMap;
+    }
+
+    private Set<String> getContextNamesToRequest(CsaProfileKeywords csaProfileKeyword) {
+        if (keywordMap.containsKey(csaProfileKeyword.toString())) {
+            return keywordMap.get(csaProfileKeyword.toString());
+        }
+        return tripleStoreCsaProfileCrac.contextNames();
```

Comment:
Are you sure you want to request on all files if there's no file associated to a keyword?